onnxruntime/core/framework/kernel_registry_manager.h

```diff
@@ -41,6 +42,8 @@ class KernelRegistryManager {
   // Then B > A > providers
   void RegisterKernelRegistry(std::shared_ptr<KernelRegistry> kernel_registry);
 
+  void RegisterKernelRegistry(std::shared_ptr<CustomRegistry> custom_registry);
```

Comment:
Why do we need this?

The kernel registry manager shouldn't need to know or care about CustomRegistry. InferenceSession should extract the kernel registry from CustomRegistry not kernel registry manager (and that's what the code appears to do).